src/renderer/controllers/prefs-controller.js

```diff
@@ -41,6 +42,9 @@ module.exports = class PrefsController {
   // All unsaved prefs take effect atomically, and are saved to config.json
   save () {
     var state = this.state
+    if (state.unsaved.prefs.isFileHandler !== state.saved.prefs.isFileHandler) {
+      ipcRenderer.send('setDefaultFileHandler', state.unsaved.prefs.isFileHandler)
+    }
```

Comment:
This is fine, but it would be better if all preferences took effect immediately instead of that the time the preferences page is unloaded/destroyed. That's how it works in all macOS apps -- and also the reason why preferences pages on macOS never have a Save/Cancel button.